pkg/apis/velero/v1/restore.go

```diff
@@ -80,6 +83,100 @@ type RestoreSpec struct {
 	// +optional
 	// +nullable
 	IncludeClusterResources *bool `json:"includeClusterResources,omitempty"`
+
+	// Hooks represent custom behaviors that should be executed during or post restore.
+	// +optional
+	Hooks RestoreHooks `json:"hooks,omitempty"`
+}
+
+// RestoreHooks contains custom behaviors that should be executed during or post restore.
+type RestoreHooks struct {
+	Resources []RestoreResourceHookSpec `json:"resources,omitempty"`
+}
+
+// RestoreResourceHookSpec defines one or more RestoreResrouceHooks that should be executed based on
+// the rules defined for namespaces, resources, and label selector.
+type RestoreResourceHookSpec struct {
+	// Name is the name of this hook.
+	Name string `json:"name"`
+
+	// IncludedNamespaces specifies the namespaces to which this hook spec applies. If empty, it applies
+	// to all namespaces.
+	// +optional
+	// +nullable
+	IncludedNamespaces []string `json:"includedNamespaces,omitempty"`
+
+	// ExcludedNamespaces specifies the namespaces to which this hook spec does not apply.
+	// +optional
+	// +nullable
+	ExcludedNamespaces []string `json:"excludedNamespaces,omitempty"`
+
+	// IncludedResources specifies the resources to which this hook spec applies. If empty, it applies
+	// to all resources.
+	// +optional
+	// +nullable
+	IncludedResources []string `json:"includedResources,omitempty"`
+
+	// ExcludedResources specifies the resources to which this hook spec does not apply.
+	// +optional
+	// +nullable
+	ExcludedResources []string `json:"excludedResources,omitempty"`
+
+	// LabelSelector, if specified, filters the resources to which this hook spec applies.
+	// +optional
+	// +nullable
+	LabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty"`
+
+	// PostHooks is a list of RestoreResourceHooks to execute during and after restoring a resource.
+	// +optional
+	PostHooks []RestoreResourceHook `json:"post,omitempty"`
```

Comment:
Can we call the JSON field `postHooks` too?

Reply:
fixed.